Automattic/jetpack
# Update assistant recommendation site type question

#### Summary

This PR updates the first question in the Jetpack Assistant to provide more useful data on the backend. It also adds a new "agency" onboarding recommendation when one of the new answers are selected (similar to how the woocommerce one works)

The reason I did not split this up into 2 PR's is the updated question is directly related to the new recommendation we are creating.

#### Changes proposed in this Pull Request:

Update site-type assistant question to more clearly differentiate between site builders and personal site owners (agencies and non-agencies). New answers to this question are `I build or manage this site for a client`, `This is an e-commerce site`, and `This is my personal site`.
Selecting `I build or manage this site for a client` will lead them to the new agencies recommendation.
Selecting `This is an e-commerce site` will lead them to the woocommerce recommendation just as before. 
Selecting both will show the agencies recommendation and then the woocommerce recommendation.
Selecting `This is my personal site` will not do anything

#### Other information:


#### Jetpack product discussion

P2: pbNhbs-3pg-p2

#### Does this pull request change what data or activity we track or use?

This makes changes to the data tracked when the assistant question is answered. Before it would save it like this:
```
answers: {
   personal: boolean
   business: boolean
   store: boolean
   other: boolean
}
```

it now tracks it as
```
answers: {
   personal: boolean
   builder: boolean
   store: boolean
}
```

All recommendations have track events when they are viewed, interacted with, or skipped. So adding the new "agencies" recommendation will inherently add more track events for it.

#### Testing instructions:

Please see the figma mockup to see what the expectation is: za5e9TEFrG8sqpPJgVaCq9-fi-22485%3A46507
Also the Google Doc for wording: https://href.li/?https://docs.google.com/document/d/1rncufOjUYCt1PCgnjsIo4JDwTnRZx0gMkb7v2GUao9s

1. Check out this branch via `git switch update/assistant-recommendation-site-type`
2. Navigate to `/wp-admin/admin.php?page=jetpack#/dashboard`
3. Reset your Jetpack Options by click on the `Reset Options (dev only)` button on the page
![Screen Shot 2022-09-22 at 2 07 58 PM](https://user-images.githubusercontent.com/65001528/191841392-863d69cf-142c-484d-8201-c47773dfb55c.jpg)
4. Refresh the page to ensure the options have been fully reset.
5. Navigate to `/wp-admin/admin.php?page=jetpack#/recommendations/site-type`
6. Select `This is my personal site`
![image](https://user-images.githubusercontent.com/65001528/191843282-e5f8d0a5-1972-4b04-99d8-6c9df991337c.png)
7. Select `Continue`, it should skip straight to the Downtime Monitoring recommendation
![image](https://user-images.githubusercontent.com/65001528/191843434-9cfc25dd-5874-43c3-bf3e-03810c1d0838.png)
8. Go back to `/wp-admin/admin.php?page=jetpack#/recommendations/site-type`
9. This time, select `I build or manage this site for a client`
10. Select `Continue` and this time you should get the new Agencies recommendation
![image](https://user-images.githubusercontent.com/65001528/191843717-14bbf6e3-c1ac-41d5-9634-4343693b7a07.png)
11. While here, test out the CTA and links on this recommendations. You can find the expected links and content in the figma file and google doc listed at the beginning of this page. The `Learn More` link should go to https://jetpack.com/for/agencies and the CTA should link to https://cloud.jetpack.com/agency/signup (NOTE: If you are already listed as an agency, this will just redirect to your agency dashboard)
12. Click `Not now` on this recommendation and you should be directed to the next recommendation. Continue selecting `Not now` until you reach the summary page
![image](https://user-images.githubusercontent.com/65001528/191845879-c0465ed1-8639-48a3-8ae4-e1484e9f2d63.png)
13. While here, click the `Sign Up` external link next to the `Jetpack for Agencies` recommendation in the list. This should again lead you to https://cloud.jetpack.com/agency/signup.
14. Now click on `Jetpack for Agencies` and confirm it takes you back to the recommendation
![image](https://user-images.githubusercontent.com/65001528/191846082-a3a1382e-c245-48f7-90c4-548d3dd93a53.png)
15. Lastly, go back to `/wp-admin/admin.php?page=jetpack#/recommendations/site-type` and select both `I build or manage this site for a client` and `This is an e-commerce site` and press `Continue`.  
![image](https://user-images.githubusercontent.com/65001528/191846328-10166677-29d5-409a-9cc1-2a112d413d29.png)
17. You should see the Agencies recommendation first. Press `Not now` 
![image](https://user-images.githubusercontent.com/65001528/191846388-265cf66b-1339-43e1-af8f-f4d0157c992f.png)
18. You should now see the woocommerce recommendation
![image](https://user-images.githubusercontent.com/65001528/191846511-b53e1c5d-9339-4d52-806e-2e3c960ea045.png)

Well that's about it!
